cpp/src/Ice/InputStream.cpp

```diff
@@ -2512,7 +2515,10 @@ Ice::InputStream::EncapsDecoder11::skipSlice()
         }
         else
         {
-            throw UnknownUserException(__FILE__, __LINE__, _current->typeId);
+            throw MarshalException{
```

Comment:
I think the previous code was bogus. UnknownUserException is not the right exception when we can't decode a user exception.

Reply:
We have this bug in all mappings, it would be good to add a test.

Reply:
Created an issue.